go/types/opcache.go

```diff
@@ -8,15 +8,55 @@ import (
 	"encoding/binary"
 	"io/ioutil"
 	"os"
+	"sync/atomic"
 
 	"github.com/attic-labs/noms/go/d"
 	"github.com/attic-labs/noms/go/hash"
 	"github.com/syndtr/goleveldb/leveldb"
 	"github.com/syndtr/goleveldb/leveldb/iterator"
 	"github.com/syndtr/goleveldb/leveldb/opt"
+	"github.com/syndtr/goleveldb/leveldb/util"
 )
 
-func newOpCache(vrw ValueReadWriter) *opCache {
+const uint32Size = 4
+
+type opCacheStore interface {
+	opCache() opCache
+	destroy() error
+}
+
+type opCache interface {
+	MapSet(mapKey Value, mapVal Value)
+	SetInsert(val Value)
+	NewIterator() opCacheIterator
+}
+
+type opCacheIterator interface {
+	MapOp() sequenceItem
+	SetOp() sequenceItem
+	Next() bool
+	Release()
+}
+
+type ldbOpCacheStore struct {
+	ldb          *leveldb.DB
+	dbDir        string
+	collectionId uint32
+	vrw          ValueReadWriter
+}
+
+type ldbOpCache struct {
+	vrw   ValueReadWriter
+	colId uint32
+	store *ldbOpCacheStore
```

Comment:
As far as I can tell, the only thing you use `store` for is to get `store.ldb`, so why not just give ldbOpCache a `*leveldb.DB` instead?